### DISCUSSION TITLE
read midi file

### DISCUSSION THREAD

**Discussion by sugizo** [Rainbow-Dreamer/musicpy#64] [Q&A]
how to read midi file ?

e.g.
pretty_midi
![image](https://github.com/Rainbow-Dreamer/musicpy/assets/2418295/07d394e9-6f91-46f6-8738-0d2b3a3d20b0)

mido
![image](https://github.com/Rainbow-Dreamer/musicpy/assets/2418295/3fd6ebec-bd53-4275-8860-cf4c76570b90)

miditoolkit
![image](https://github.com/Rainbow-Dreamer/musicpy/assets/2418295/dd90af9b-c6ea-4941-8974-d198478795e7)

thanks and best regards

**Comment by Rainbow-Dreamer** [comment]
Hi, you can read musicpy's wiki about how to read midi file, here is the link to the chapter: https://github.com/Rainbow-Dreamer/musicpy/wiki/Useful-functionality#read-midi-files-and-convert-them-to-piece-types-for-various-music-theory-operations
